src/Node/Express/Handler.purs

```diff
@@ -37,283 +39,271 @@ import Node.Express.Internal.Response
 import Node.Express.Internal.Request
 import Node.Express.Internal.QueryString
 
---| Monad responsible for handling single request.
-data HandlerM a = HandlerM (Request -> Response -> ExpressM Unit -> ExpressM a)
-type Handler = HandlerM Unit
+-- | Monad responsible for handling single request.
+data HandlerM e a = HandlerM (Request -> Response -> Eff e Unit -> Aff e a)
 
+type Handler e = HandlerM (express :: EXPRESS | e) Unit
 
-instance functorHandlerM :: Functor HandlerM where
+instance functorHandlerM :: Functor (HandlerM e) where
     map f (HandlerM h) = HandlerM \req resp nxt ->
         (h req resp nxt >>= \r -> return $ f r)
 
-instance applyHandlerM :: Apply HandlerM where
+instance applyHandlerM :: Apply (HandlerM e) where
     apply (HandlerM f) (HandlerM h) = HandlerM \req resp nxt -> do
         res   <- h req resp nxt
         trans <- f req resp nxt
         return $ trans res
 
-instance applicativeHandlerM :: Applicative HandlerM where
+instance applicativeHandlerM :: Applicative (HandlerM e) where
     pure x = HandlerM \_ _ _ -> return x
 
-instance bindHandlerM :: Bind HandlerM where
+instance bindHandlerM :: Bind (HandlerM e) where
     bind (HandlerM h) f = HandlerM \req resp nxt -> do
         (HandlerM g) <- liftM1 f $ h req resp nxt
         g req resp nxt
 
-instance monadHandlerM :: Monad HandlerM
+instance monadHandlerM :: Monad (HandlerM e)
 
-instance monadEffHandlerM :: MonadEff eff HandlerM where
-    liftEff act = HandlerM \_ _ _ -> unsafeInterleaveEff act
+instance monadEffHandlerM :: MonadEff eff (HandlerM eff) where
+    liftEff act = HandlerM \_ _ _ -> liftEff act
 
+instance monadAffHandlerM :: MonadAff eff (HandlerM eff) where
+    liftAff act = HandlerM \_ _ _ -> act
 
-withHandler :: forall a. HandlerM a -> Request -> Response -> ExpressM Unit -> ExpressM a
-withHandler (HandlerM h) = h
+runHandlerM :: forall e a. HandlerM (express :: EXPRESS | e) a -> Request -> Response -> ExpressM e Unit -> ExpressM e Unit
```

Comment:
Since this function returns `Unit` anyway, I think its singnature should use `Handler e` instead of `HandlerM (express :: EXPRESS | e) a`